packages/react-router-dev/typegen/generate.ts

```diff
@@ -346,3 +348,49 @@ function paramsType(path: string) {
     })
   );
 }
+
+// https://github.com/remix-run/react-router/blob/7a7f4b11ca8b26889ad328ba0ee5a749b0c6939e/packages/react-router/lib/router/utils.ts#L894C1-L937C2
```

Comment:
have to copy from `react-router`. is there some way to not duplicate this w/o pulling in `react-router` as a dependency? maybe list `react-router` as a peer dep